beacon-chain/core/helpers/legacy.go

```diff
@@ -0,0 +1,21 @@
+package helpers
+
+import (
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
+)
+
+// DepositRequestsHaveStarted determines if the deposit requests have started
+func DepositRequestsHaveStarted(beaconState state.BeaconState) bool {
+	if beaconState.Version() >= version.Electra {
+		requestsStartIndex, err := beaconState.DepositRequestsStartIndex()
+		if err == nil {
+			// deposit_requests_start_index will only be set once
+			// eth1data will be frozen
+			if beaconState.Eth1DepositIndex() == requestsStartIndex {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
```suggestion
// HasDepositRequestStarted determines if the deposit requests have started.
func HasDepositRequestStarted(beaconState state.BeaconState) bool {
    if beaconState.Version() < version.Electra {
        return false
    }

    requestsStartIndex, err := beaconState.DepositRequestsStartIndex()
    if err != nil {
        return false
    }

    return beaconState.Eth1DepositIndex() == requestsStartIndex
}
```

I'm not sure which function name will be appropriate, but this could remove nested if-else code blocks.

Reply:
yeah that's a good point let's update it 😓